### ISSUE TITLE
"Unhandled exception in concurrent task" repeats in log after CLN 24.05 upgrade

### ISSUE THREAD

**Issue by aaroncarlucci** [ksedgwic/clboss#213]
After upgrading to CLN `24.05`, running clboss `v0.13.1`, I'm seeing the following entries in the CLN logs repeating exactly every 10 minutes:
```
Jun 06 09:04:47  lightningd[510666]: Unhandled exception in concurrent task!
Jun 06 09:04:47  lightningd[510666]: Incorrect type.
Jun 06 09:04:47  lightningd[510666]: ...main loop continuing...
```

Set the CLN `log-level=io` hoping to get more information, but that didn't help. @ksedgwic pointed to the following on Discord:

https://github.com/ZmnSCPxj/clboss/blob/df51d5486b05acbeca16cb9e42d18fa082e6079a/Ev/concurrent.cpp#L19-L29

I'm happy to provide more information if provided with tips to extract more data.

**Comment by ksedgwic** [comment]
The 10 minutes suggests it's one of:
```
ChannelFinderByPopularity.cpp:181:		bus.subscribe< Msg::Timer10Minutes
ListfundsAnnouncer.cpp:24:	bus.subscribe< Msg::Timer10Minutes
ListpeersAnnouncer.cpp:51:	bus.subscribe<Msg::Timer10Minutes
OnchainFeeMonitor.cpp:342:		bus.subscribe<Msg::Timer10Minutes>([this](Msg::Timer10Minutes const&) {
SelfUptimeMonitor.cpp:71:		bus.subscribe<Msg::Timer10Minutes
SendpayResultMonitor.cpp:130:		bus.subscribe<Msg::Timer10Minutes
SwapManager.cpp:71:		bus.subscribe<Msg::Timer10Minutes
TimerTwiceDailyAnnouncer.cpp:34:		bus.subscribe<Msg::Timer10Minutes
PeerComplaintsDesk/Main.cpp:143:		bus.subscribe<Msg::Timer10Minutes
```

**Comment by ksedgwic** [comment]
There's a good chance that https://github.com/ZmnSCPxj/clboss/pull/198 will fix this, it fixes a large number of code locations which were using a now deprecated API version ...

How risk adverse are you with this node?  The above fix and some others are in `master`.  I've been happily running on several testnet instances w/o problem.

**Comment by aaroncarlucci** [comment]
@ksedgwic It's a production node, with some value, but I like to live on the edge every now and then. You suggest I consider running the `master` branch directly, then?

**Comment by vincenzopalazzo** [comment]
I am running clnboss on my nodes without any problem, but I think I am running the master with the https://github.com/ZmnSCPxj/clboss/pull/198 included

>You suggest I consider running the master branch directly, then?

Why you should not @aaroncarlucci ? For CLN is always good run the master IMHO

**Comment by aaroncarlucci** [comment]
@ksedgwic @vincenzopalazzo I've joined with you in the ranks of frontline warriors running `master`. So far, the startup BROKEN log entry that I pasted Ken in Discord has disappeared, and after 20 minutes uptime, no log messages as reported in the original issue description.

I'll check in on it  tomorrow AM to be sure, but things look encouraging so far. Not sure if this is the behavior for other users after upgrading to CLN `24.05`, but IMHO, it might be time to consider another point release.

Thanks for the support!
